core/src/main/scala/besom/internal/codecs.scala

```diff
@@ -1074,17 +1075,25 @@ object Encoder:
         case a: Asset   => assetEncoder.encode(a)
         case a: Archive => archiveEncoder.encode(a)
 
-  given listEncoder[A](using innerEncoder: Encoder[A]): Encoder[List[A]] = new Encoder[List[A]]:
-    def encode(list: List[A])(using Context): Result[(Metadata, Value)] =
+  given iterableEncoder[A](using innerEncoder: Encoder[A]): Encoder[Iterable[A]] = new Encoder[Iterable[A]]:
+    def encode(iterable: Iterable[A])(using Context): Result[(Metadata, Value)] =
       Result
-        .sequence(list.map(innerEncoder.encode(_)))
+        .sequence(iterable.map(innerEncoder.encode(_)))
         .map { lst =>
           val (resources, values) = lst.unzip
           val joinedMetadata      = resources.foldLeft(Metadata.empty)(_.combine(_))
 
-          joinedMetadata -> values.asValue
+          joinedMetadata -> values.toList.asValue
         }
 
+  given listEncoder[A](using innerEncoder: Encoder[A]): Encoder[List[A]] = iterableEncoder.contramap[List[A]](a => a)
+
+  given seqEncoder[A](using innerEncoder: Encoder[A]): Encoder[Seq[A]] = iterableEncoder.contramap[Seq[A]](a => a)
+
+  given indexedSeqEncoder[A](using innerEncoder: Encoder[A]): Encoder[IndexedSeq[A]] = iterableEncoder.contramap[IndexedSeq[A]](a => a)
+
+  given setEncoder[A](using innerEncoder: Encoder[A]): Encoder[Set[A]] = iterableEncoder.contramap[Set[A]](a => a)
```

Comment:
This instance is broken for any situation where an element of the set has the same encoded representation because in `interableEncoder` when you `map` on `iterable` that iterable is a `Set` underneath. It's a gotcha with Sets being subtypes of iterables that have a `map` function because for sets the size of resulting collection is between 1 and n:
```
scala> def test(it: Iterable[Int]): Iterable[String] = it.map(_ => "oops")
def test(it: Iterable[Int]): Iterable[String]

scala> test(Set(1, 2, 3)).size
val res0: Int = 1
```
To make it safe you should sequence on a list in `iterableEncoder` so `Result.sequence(iterable.toList.map(...`